module-system/sov-state/src/prover_storage.rs

```diff
@@ -79,11 +79,11 @@ impl<S: MerkleProofSpec> Storage for ProverStorage<S> {
             .map(|root| root.0)
     }
 
-    fn validate_and_commit(
+    fn compute_state_update(
         &self,
         state_accesses: OrderedReadsAndWrites,
         witness: &Self::Witness,
-    ) -> Result<[u8; 32], anyhow::Error> {
+    ) -> Result<([u8; 32], NodeBatch), anyhow::Error> {
```

Comment:
Not necessarily a blocker, but it's unfortunate that we leak the `NodeBatch` type into the interface now. Up until now, the `Storage` trait was decoupled from the jmt crate (IIRC), but NodeBatch is a jmt type.

This isn't a huge deal, because we our *database* is jmt specific anyway - so we would always have had to fix that before adding another `Storage` implementation - but this is moving us a little further from the idea end state.

Reply:
Very good point. Added `StateUpdate` associated type to the `Storage`.